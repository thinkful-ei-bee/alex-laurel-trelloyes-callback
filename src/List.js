import React from 'react';
import Card from './Card';
import './List.css';

export default function List(props){
    return (
        <section className='List'>
        <header className='List-header'>
          <h2>{props.header}</h2>
        </header>
        <div className="List-cards">
          {props.cards.map((item, index)=>
            <Card key ={index} title ={item.title} content={item.content}/>
          )}
        
        <button type='button' className='randoCard'>+ Add Random Card</button>
        </div>
      </section>
    )
}