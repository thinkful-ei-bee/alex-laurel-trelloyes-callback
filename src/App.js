import React, { Component } from 'react';
import List from './List'
import './App.css';

class App extends Component {
   store={
     lists:[],
     allCards:[]
   }

  render() {
    let {store} =this.props;
    return (
      <main className='App'>
        <header className='App-header'>
          <h1>Trelloyes!</h1>
        </header>
        <div className='App-list'>
          {store.lists.map((items=> <List key ={items.id} header ={items.header} 
          cards={items.cardIds.map(itemList=>store.allCards[itemList])}/>
            ))}
        </div>
      </main>
    );
  }
}

export default App;
