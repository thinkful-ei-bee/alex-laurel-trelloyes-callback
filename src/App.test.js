import React from 'react';
import ReactDOM from 'react-dom';
import store from './store';
import List from './List';
import Card from './Card';
import App from './App';
import renderer from 'react-test-renderer';

describe('My App test', ()=>{
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App store={store}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected',()=>{
    const tree = renderer.create(<List key={store.lists[0].id} header ={store.lists[0].header} 
      cards={store.lists[0].cardIds.map(itemList=>store.allCards[itemList])}/> ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the UI as expected',()=>{
    const tree = renderer.create(<Card key ={'a'} title ={store.allCards.a.title} content={store.allCards.a.content}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
  