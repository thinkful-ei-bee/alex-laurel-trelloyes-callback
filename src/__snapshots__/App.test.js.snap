// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My App test renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Twelth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="randoCard"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`My App test renders the UI as expected 2`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    First card
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;
